src/components/movie-page/movie-page.test.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+import MoviePage, {generateRatingStr, getRatingPlural, getActorsStringWithPostfix} from './movie-page.jsx';
+import promoObj from '../../mocks/promo.js';
+
+describe(`MoviePage Component`, () => {
+  it(`renders`, () => {
+    const tree = renderer.create(
+        <MoviePage movieData={promoObj} />
+    ).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+});
+
+describe(`MoviePage utils`, () => {
+  it(`fn generateRatingStr works correctly`, () => {
+    const reference = {
+      bad: `Bad`,
+      normal: `Normal`,
+      good: `Good`,
+      veryGood: `Very good`,
+      awesome: `Awesome`
+    };
+
+    expect(generateRatingStr(1)).toEqual(reference.bad);
+    expect(generateRatingStr(3)).toEqual(reference.normal);
+    expect(generateRatingStr(6)).toEqual(reference.good);
+    expect(generateRatingStr(9.99)).toEqual(reference.veryGood);
+    expect(generateRatingStr(10)).toEqual(reference.awesome);
```

Comment:
Сперва хотел написать тест, который покрывает все возможные входные значение, но потом решил, что это overkill.
Скажи, на реальных проектах проводятся такие всепокрывающие тесты?

```suggestion
    const fnCoverageGenerator = (from, to) => {
      let valuesArr = [];
      let trimmedResult = [];
      for (let i = from; i < to; i += 0.01) {
        valuesArr.push(+i.toFixed(3));
      }

      valuesArr.forEach((it) => {
        const currentVal = generateRatingStr(it);
        if (!trimmedResult.includes(currentVal)) {
          trimmedResult.push(currentVal);
        }
      });

      return trimmedResult;
    };

    expect(generateRatingStr(2)).toEqual(reference.bad);
    expect(fnCoverageGenerator(0, 2.99)).toHaveLength(1);

    expect(generateRatingStr(4)).toEqual(reference.normal);
    expect(fnCoverageGenerator(3, 4.99)).toHaveLength(1);

    expect(generateRatingStr(5)).toEqual(reference.good);
    expect(fnCoverageGenerator(5, 7.99)).toHaveLength(1);

    expect(generateRatingStr(9)).toEqual(reference.veryGood);
    expect(fnCoverageGenerator(8, 9.99)).toHaveLength(1);

    expect(generateRatingStr(10)).toEqual(reference.awesome);
```

Reply:
```
const generateRatingStr = (int) => {
  if (int < 3) {
    return `Bad`;
  } else if (int >= 3 && int < 5) {
    return `Normal`;
  } else if (int >= 5 && int < 8) {
    return `Good`;
  } else if (int >= 8 && int < 10) {
    return `Very good`;
  } else {
    return `Awesome`;
  }
};
```
 ты должен был в тесте нга эту функцию покрывтаь все варианты ветвления функции, и этого достаточно